Modules/+Sources/Msquared.m

```diff
@@ -0,0 +1,481 @@
+            obj.PulseBlaster.lines(obj.PB_line).state = val;
+        end
+        function val = set_armed(obj, val, ~)       % Checks if the laser is outputting power (i.e. has been armed) and complains if result was contradictory.
+            obj.getFrequency(); 
+            
+            if val ~= (obj.output_monitor > .01)    % If we are setting armed to an incorrect value...
+                val = ~val;
+                if val
+                    warndlg(['Request to arm laser. Laser is off, as output_monitor reads ' num2str(obj.output_monitor) '; please turn the laser on'], 'Arm (Sources.Msquared)')
+                else
+                    warndlg(['Request to blackout laser. Laser is on, as output_monitor reads ' num2str(obj.output_monitor) '; please turn the laser off'], 'Blackout (Sources.Msquared)')
+                end
+            end
+        end
+        
+        function host= set_hwserver_host(obj, host, ~)
+            try
+                obj.hwserver = hwserver(host); %#ok<CPROPLC>
+                
+                % Update laser list. (This piece of code was breaking due to update_settings trying to update hwserver_host, which is in use. Instead hardcode moduleName options.)
+%                 opts = obj.hwserver.get_modules('msquared.');
+%                 mp = obj.get_meta_pref('moduleName');
+%                 if ~isequal(mp.choices,opts) % Only update if different
+%                     if ~isempty(mp.value)
+%                         % If we need to reset it, we also need to re-grab the metapref
+%                         obj.moduleName = '';
+%                         mp = obj.get_meta_pref('moduleName');
+%                     end
+%                     mp.choices = opts;
+%                     obj.set_meta_pref('moduleName', mp);
+%                     notify(obj,'update_settings');
+%                 end
+            catch
+                host = Sources.Msquared.no_server;
+                obj.active_module = host;
+                obj.status = host;
+            end
+        end
+        function val = set_moduleName(obj,val,~)
+            obj.getFrequency();
+        end
+        
+        function val = set_refresh(obj,val,~)       % Polls hwserver by calling getFrequency().
+            obj.getFrequency();
+        end
+        
+        function val = set_PB_line(obj,val,~)
+            if ~isempty(obj.PulseBlaster)
+                obj.source_on = obj.PulseBlaster.lines(obj.PB_line).state;
+            end
+        end
+        function host= set_PB_host(obj,host,~)
+            try
+                obj.PulseBlaster = Drivers.PulseBlaster.instance(host);
+                obj.source_on = obj.PulseBlaster.lines(obj.PB_line).state;
+            catch
+                host = obj.no_server;
+            end
+        end
+        
+        function val = set_target_wavelength(obj, val, pref)   % Calls the tune() method.
+            if isnan(val); return; end % Short circuit on NaN
```

Comment:
What's the usecase for this being NaN?

Reply:
Upon startup, the default value of setpoint_ is NaN. Also, I think this behavior is fine. NaN doesn't cause an error, and nothing happens.